pkg/operator/controller/externaldns/pod.go

```diff
@@ -325,30 +328,33 @@ func (b *externalDNSContainerBuilder) fillAWSFields(container *corev1.Container)
 
 	env := []corev1.EnvVar{
 		{
-			Name: awsAccessKeyIDEnvVar,
+			Name: awsSharedCredentialsFile,
 			ValueFrom: &corev1.EnvVarSource{
 				SecretKeyRef: &corev1.SecretKeySelector{
```

Comment:
The `AWS_SHARED_CREDENTIALS_FILE` environment variable should contain the path to the credentials file. Over here the variable contains the contents of the file. Not it's path.

Reply:
Updated the correct env var value